lib/OpenQA/WebAPI/Plugin/ObsRsync/Task.pm

```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 2019 SUSE Linux GmbH
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package OpenQA::WebAPI::Plugin::ObsRsync::Task;
+use Mojo::Base -strict;
+use IPC::System::Simple qw(system $EXITVAL);
+
+my $lock_timeout = 36000;
+
+sub _run {
+    my ($app, $job, $args) = @_;
+    my $project        = $args->{project};
+    my $home           = $app->obs_rsync->home;
+    my $concurrency    = $app->obs_rsync->concurrency;
+    my $retry_interval = $app->obs_rsync->retry_interval;
+    my $queue_limit    = $app->obs_rsync->queue_limit;
+
+    if ($job->info && !$job->info->{notes}{project_lock}) {
+        return $job->retry({delay => $retry_interval})
+          unless $app->minion->lock("obs_rsync_project_" . $project . "_lock", $lock_timeout);
+
+        $job->note(project_lock => 1);
+    }
```

Comment:
I don't particularly like this and believe a simple guard would work just as well (if not better). But the logic is simple enough that it's no longer a blocker for me.